GitCommands/DiffListSortService.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GitCommands
+{
+    public class DiffListSortService : IDiffListSortService
+    {
+        private static readonly Lazy<DiffListSortService> _lazyDiffListSorting = new Lazy<DiffListSortService>(() => new DiffListSortService());
+
+        public static DiffListSortService Instance => _lazyDiffListSorting.Value;
+
+        private DiffListSortType _diffListSorting;
+
+        public DiffListSortType DiffListSorting
+        {
+            get { return _diffListSorting; }
+            set
+            {
+                if (value == DiffListSortType.Invalid)
+                {
+                    throw new ArgumentException("Cannot use invalid sorting.");
+                }
+
+                var previous = _diffListSorting;
+                if (previous != value)
+                {
+                    _diffListSorting = value;
+                    AppSettings.DiffListSorting = value;
+                    RaiseSortChange();
+                }
+            }
+        }
+
+        public DiffListSortService()
+        {
+            DiffListSorting = GetSettingValueOrDefault();
+        }
+
+        private DiffListSortType GetSettingValueOrDefault()
+        {
+            var persistedSetting = AppSettings.DiffListSorting;
+            if (persistedSetting == DiffListSortType.Invalid)
+            {
+                return DiffListSortType.FilePath;
+            }
+
+            return persistedSetting;
```

Comment:
I'd move this block into `AppSettings` and delegate the responsibility for providing a correct value to `DiffListSorting` implementation

Reply:
I put this code here so that it was easier to test. Is there a way to adequately test the AppSettings class given that there is an invalid enum value?